csp/middleware.py

```diff
@@ -95,7 +100,8 @@ def process_response(self, request: HttpRequest, response: HttpResponseBase) ->
         # Once we've written the header, accessing the `request.csp_nonce` will no longer trigger
         # the nonce to be added to the header. Instead we throw an error here to catch this since
         # this has security implications.
-        setattr(request, "csp_nonce", SimpleLazyObject(self._csp_nonce_post_response))
+        if getattr(request, "_csp_nonce", None) is None:
+            setattr(request, "csp_nonce", FalseLazyObject(self._csp_nonce_post_response))
```

Comment:
This is a significant change...

- If the nonce was used during the request it should then be set and not `None`. In which case, we could continue to read it after the middleware just fine.
- If the none was NOT used during the request, we update it with this so a boolean check returns False and any attempt to access it raises the `CSPNonceError`.

That seems like a nice approach. We should document this change in behavior.

Reply:
I agree, I'll update `docs/nonce.rst`. I think that would be a good place for my advice for middleware that needs the nonce.